packages/suite-desktop-core/src/hang-detect.ts

```diff
@@ -40,17 +43,22 @@ export const hangDetect = (
         timeout = setTimeout(timeoutCallback, HANG_WAIT);
         ipcMain.handleOnce('handshake/client', () => {
             clearTimeout(timeout);
-            // always resolve repeated handshakes from renderer (e.g. Ctrl+R)
-            ipcMain.handle('handshake/client', ipcEvent => {
-                validateIpcMessage(ipcEvent);
-
-                return Promise.resolve({});
-            });
+            try {
+                // always resolve repeated handshakes from renderer (e.g. Ctrl+R)
+                ipcMain.handle('handshake/client', handshakeHandler);
```

Comment:
Now it's possible to enter this code block more than once? The logic behind it was that the main-side handshake runs only once (therefore handleOnce) while renderer-side handshakes are sent on every Ctrl+R and main just returns empty response. Is this still valid?